private/pkg/storage/multi.go

```diff
@@ -76,6 +76,15 @@ func newMultiReadBucket(
 }
 
 func (m *multiReadBucket) Get(ctx context.Context, path string) (ReadObjectCloser, error) {
+	if m.overlay {
+		// If overlay is enabled, attempt a Get operation against the first bucket. If the file is
+		// found, we avoid the potentially expensive Stat call.
+		if readObjectCloser, err := m.delegates[0].Get(ctx, path); err != nil && !IsNotExist(err) {
+			return nil, err
+		} else if readObjectCloser != nil {
```

Comment:
I don't think this is right, I think you want

if err != nil {
  if !IsNotExist(err) {
    return nil, err
  }
  return nil, nil
}
return readObjectCloser, nil

Reply:
edit: nevermind

Reply:
Scanned the codebase a bit, and updated the logic to match the typical style of handling the error within the error block and adding an else otherwise.